tools/sof-perf-analyzer.py

```diff
@@ -0,0 +1,253 @@
+                span_end_pos = match_obj.span()[1]
+                line_split = line[span_end_pos + 1:].split()
+                widget_name = line_split[0]
+                # In the linux kernel, IDA is used to allocated pipeline widget instance ID,
+                # this ID later is used for pipeline creation, thus becomes pipeline ID in the
+                # firmware. Note that ppl_id variable will be assigned properly at pipeline widget
+                # creation, because it is always the first one to be created before all other
+                # widgets in the same pipeline and pipelines are created sequentially.
+                if widget_name.startswith('pipeline'):
+                    ppl_id = line_split[2]
+                next_line = next(f)
+                widget_id = next_line.split('|')[0].split(':')[-1].strip()
+                # convert to the same ID format in mtrace
+                widget_id = '0x' + widget_id[:-6:-1]
+                comp_name_map[Comp(ppl_id, widget_id)] = widget_name
+
+def analyze_perf_info():
+    '''Calculate performance statics from performance information'''
+    for comp, perf_info_list in perf_info.items():
+        len_perf_info = len(perf_info_list)
+        avg_cpu_avg = sum(e.cpu_avg for e in perf_info_list) / len_perf_info
+        avg_cpu_peak = sum(e.cpu_peak for e in perf_info_list) / len_perf_info
+        peak_cpu_avg = max(e.cpu_avg for e in perf_info_list)
+        peak_cpu_peak = max(e.cpu_peak for e in perf_info_list)
+        perf_stats[comp] = CompPerfStats(avg_cpu_avg, avg_cpu_peak, peak_cpu_avg, peak_cpu_peak)
+
+def print_perf_info():
+    '''Output SOF trace performance statics'''
+    if len(perf_stats):
+        title_fmt = '{:>30}  {:10}  {:<15}  {:<15}  {:<15}  {:<15}'
+        title = title_fmt.format('COMP_NAME', 'COMP_ID', 'AVG_OF_CPU_AVG', 'AVG_OF_CPU_PEAK',
+                             'PEAK_OF_CPU_AVG', 'PEAK_OF_CPU_PEAK')
+        print(title)
+
+        stats_fmt = '{:>30}  {:10}  {:<15,.2f}  {:<15,.2f}  {:<15,.2f}  {:<15,.2f}'
+        for comp, perf in perf_stats.items():
+            avg_cpu_avg_mcps = perf.avg_cpu_avg * TICK_TO_MCPS
+            avg_cpu_peak_mcps = perf.avg_cpu_peak * TICK_TO_MCPS
+            peak_cpu_avg_mcps = perf.peak_cpu_avg * TICK_TO_MCPS
+            peak_cpu_peak_mcps = perf.peak_cpu_peak * TICK_TO_MCPS
+
+            comp_name = comp_name_map.get(comp, 'None')
+            comp_id = f'{comp.ppl_id}-{comp.comp_id}'
+            stat = stats_fmt.format(comp_name, comp_id, avg_cpu_avg_mcps, avg_cpu_peak_mcps,
+                                    peak_cpu_avg_mcps, peak_cpu_peak_mcps)
+            print(stat)
+
+def parse_args():
+    '''Parse command line arguments'''
+    parser = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+                                     description=__doc__)
+    parser.add_argument('filename')
+    parser.add_argument('--dmesg', type=pathlib.Path, required=False,
```

Comment:
Don't use the name `dmesg`. `dmesg` is just one of the ways to access `kmsg` kernel logs and it's probably the worst way because it's a limited size ring buffer where logs expire. It also has no timestamp, not hostname and no log level. Also make sure this script is compatible with `kern.log` and `journalctl -k`. Being compatible with `dmesg` is fine but it should never look like it recommends `dmesg`.

```suggestion
    parser.add_argument('--kmsg', type=pathlib.Path, required=False,
```